src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/NamingStyles/NamingStyle.cs

```diff
@@ -349,17 +349,22 @@ public static string StripCommonPrefixes(string name, out string prefix)
                     case 'm':
                     case 's':
                     case 't':
-                        if (index + 2 < name.Length && name[index + 1] == '_')
+                        if (index + 1 < name.Length && name[index + 1] == '_')
                         {
-                            index += 2;
+                            index++;
```

Comment:
Not really. Lets take name `m_0`. At the first iteration of the `while`loop `index` will be 0 (at position of `m`). The original logic checked whether there is a `_` after `index` and then made 2 steps, cutting of in our case `m_`, leading to invalid `0` name. I made it only one step, so it will cut only `m` and in the next iteration will decide whether it should eat `_` as well or not

Reply:
Ah, that works for me